docs/src/main/asciidoc/security-authorize-web-endpoints-reference.adoc

```diff
@@ -46,6 +46,36 @@ quarkus.http.auth.policy.role-policy1.roles-allowed=user,admin
 ----
 <1> This defines a role-based policy that allows users with the `user` and `admin` roles.
 
+Sometimes it might be useful to register your own policy. You can get it done by creating application scoped CDI bean
+that implements the `io.quarkus.vertx.http.runtime.security.NamedHttpSecurityPolicy` interface like in the example below:
```

Comment:
Today users can already register `HttpSecurityPoliicy` - in which case it will apply to all methods. 
I'm not sure a new interface is needed, can `HttpSecurityPoliicy` have a default `name()` returning null instead ?

By the way, it is not really related to the actual problem this PR (nicely) resolves. IMHO it just complicates things, since making it runtime only is a major change of its own, and while having a named custom policy is very interesting, IMHO it should be done independently of this PR - this way we'll have 2 real noteworthy features as well

Reply:
> Today users can already register HttpSecurityPoliicy - in which case it will apply to all methods.
I'm not sure a new interface is needed, can HttpSecurityPoliicy have a default name() returning null instead ?

I originally had exactly that, but the `PathMatchingHttpSecurityPolicy` is a global `HttpSecurityPolicy`, so injecting `HttpSecurityPolicy` and checking for `name` will lead to circular dependency. I suppose I can analyze `BeanInfo` or something like that for bean class, but it's far from elegant.

> while having a named custom policy is very interesting, IMHO it should be done independently of this PR - this way we'll have 2 real noteworthy features as well

I think you misunderstood this. It is not a new thing, previously you could do that via build item, now since the config is runtime, it doesn't make sense, so I allowed it as a CDI bean.

Reply:
Just to be clear - I don't dismiss your comment, I agree in general, but I also need to keep backward compatibility, so it needs to be done in this PR or I'll have to introduce some hack just for backwards compatibility. I'll wait to hear your opinion before I make any changes.

Reply:
Well, considering it's a `Singleton`, I'll only have to do it once, so I can check for class name. Alright, I'll get rid of `NamedHttpSecurityPolicy`.

Reply:
Done.

Reply:
Thanks @michalvavrik, sorry if I missed some of your ideas behind it, I just thought if someone already implemented `HttpSecurityPolicy` and now finds out about this new enhancement (which I absolutely support, very nice idea :+1: ) and want to apply this policy to specific paths only, they would have to change interface as opposed to introducing a name property which appears to be simpler. I think we have something similar for one of the other HTTP related security classes